### DISCUSSION TITLE
RFC: new option to ignore boosted in hxRequest

### DISCUSSION THREAD

**Discussion by resmo** [wimdeblauwe/htmx-spring-boot#138] [Ideas]
We currently have 2 endpoints, one for the hx requests and one for a full page load. 

```java
@GetMapping("/{id}")
public String detail(@PathVariable Long id, Model model) {
}

@HxRequest
@GetMapping("/{id}")
public HtmxResponse hxDetail(@PathVariable Long id, Model model) {
}
```

Now, if we use `hx-boost=true`, the endpoint for the hx-requests is used per default. 

However the purpose of hx-boost is IMHO to trigger a full page load request and let htmx handle it differently on the frontend.

We are aware of `htmxRequest.isBoosted()` but we think it would be much more useful to allow to ignore _boosted_ optionally, so the full page load can be used.

```java
@HxRequest(ignore_boosted=true) 
```

We would suggest to set the default to `false` for backwards compatibility.

**Comment by wimdeblauwe** [comment]
Seems like something that we could implement since htmx sends the `HX-Boosted` header along that we could look into. I created issue https://github.com/wimdeblauwe/htmx-spring-boot/issues/139 to further address this.

**Comment by wimdeblauwe** [comment]
This is now released as part of release 3.6.0: https://github.com/wimdeblauwe/htmx-spring-boot/releases/tag/3.6.0
